In [1]:
import os
import geopandas as gpd

In [2]:
# Define the folder containing the KFS data files
# Since the notebook is located in the "notebooks" directory/folder,
# we need to go up one level(..) to access the "data/raw/kfs" folder.

folder = "../data/raw/kfs"

# Verify that Python can locate the data folder and the files within it

# Shows where the notebook is running from
print("Current directory:", os.getcwd())  
# Checks whether the folder exists   
print("Folder exists:", os.path.exists(folder))
# Lists all files inside the folder
print("Files:", os.listdir(folder))

# List of KFS datasets to inspect

files = [
    "kfs_raw.geojson",
    "protected_area.geojson",
    "kenya_gazetted_forest_-2241218727256807108.geojson",
    "Kenya_Forest_Type_2010_Dataset.geojson"
]

# Loop through each dataset and inspect its contents

for file in files:

    # Create the full file path
    path = os.path.join(folder, file)

    # Display a separator for readability
    print("\n" + "=" * 80)

    # Print the name of the dataset being inspected
    print("Reading:", path)
    
    # Check whether the file exists before reading it
    print("Exists?", os.path.exists(path))

    # Load the GeoJSON file into a GeoDataFrame
    gdf = gpd.read_file(path)

    # Display the number of records(rows)
    print("Rows:", len(gdf))

    # Display all available columns
    print("Columns:")
    print(gdf.columns.tolist())

Current directory: c:\Users\Owner\Desktop\Forest-Explorer-Kenya\notebooks
Folder exists: True
Files: ['Kenya_Forest_Type_2010_Dataset.geojson', 'kenya_gazetted_forest_-2241218727256807108.geojson', 'kfs_raw.geojson', 'protected_area.geojson']

Reading: ../data/raw/kfs\kfs_raw.geojson
Exists? True
Rows: 219
Columns:
['id', 'AREA', 'PERIMETER', 'FORESTS_', 'FORESTS_ID', 'LU', 'CFACTOR', 'geometry']

Reading: ../data/raw/kfs\protected_area.geojson
Exists? True
Rows: 170
Columns:
['id', 'AREA', 'PERIMETER', 'PARK_POL_', 'PARK_POL_I', 'NUMBER', 'AREANAME', 'ISO3', 'SIZE', 'YEAR', 'IUCNCAT', 'LON', 'LAT', 'CNTRYNAME', 'DESIGNATE', 'IUCNUM', 'geometry']

Reading: ../data/raw/kfs\kenya_gazetted_forest_-2241218727256807108.geojson
Exists? True
Rows: 388
Columns:
['FID', 'FOREST', 'GAZETTED', 'area', 'geometry']

Reading: ../data/raw/kfs\Kenya_Forest_Type_2010_Dataset.geojson
Exists? True
Rows: 49309
Columns:
['FID', 'OBJECTID', 'DGC_CODICE', 'FORE_TY_10', 'Area', 'Area1', 'Shape_Leng', 'Shape_A

#### Deep-Diviving into the kfs_raw.geojson

##### Step 1: Load the dataset

In [3]:
import geopandas as gpd

# Load the dataset
kfs_raw = gpd.read_file("../data/raw/kfs/kfs_raw.geojson")

##### Step 2: General information

In [4]:
# Dataset dimensions
print("Shape:", kfs_raw.shape)

# Column names
print("\nColumns:")
print(kfs_raw.columns.tolist())

# Data types
print("\nData Types:")
print(kfs_raw.dtypes)

Shape: (219, 8)

Columns:
['id', 'AREA', 'PERIMETER', 'FORESTS_', 'FORESTS_ID', 'LU', 'CFACTOR', 'geometry']

Data Types:
id                 str
AREA           float64
PERIMETER      float64
FORESTS_         int32
FORESTS_ID       int32
LU                 str
CFACTOR        float64
geometry      geometry
dtype: object


##### Step 3: Preview the data

In [5]:
# Display the first 10 rows
kfs_raw.head(10)

,id,AREA,PERIMETER,FORESTS_,FORESTS_ID,LU,CFACTOR,geometry
0,ken_forests.1,0.005364,0.408104,2,372,F1,0.001,"MULTIPOLYGON (((36.91195 2.72936, 36.90744 2.7..."
1,ken_forests.2,0.000712,0.130134,3,93,F1,0.001,"MULTIPOLYGON (((35.04576 2.35031, 35.04461 2.3..."
2,ken_forests.3,0.010831,0.453610,4,509,F1,0.001,"MULTIPOLYGON (((37.97448 2.33895, 37.97785 2.3..."
3,ken_forests.4,0.000340,0.068525,5,101,F1,0.001,"MULTIPOLYGON (((35.0773 2.28476, 35.07729 2.29..."
4,ken_forests.5,0.012640,0.561492,6,360,F1,0.001,"MULTIPOLYGON (((36.7827 2.1279, 36.7827 2.1426..."
5,ken_forests.6,0.008526,0.630345,7,390,F1,0.001,"MULTIPOLYGON (((37.1842 1.69491, 37.18532 1.70..."
6,ken_forests.7,0.000764,0.131530,8,161,F1,0.001,"MULTIPOLYGON (((35.39681 1.73295, 35.39792 1.7..."
7,ken_forests.8,0.002385,0.198598,9,154,F1,0.001,"MULTIPOLYGON (((35.36876 1.67186, 35.36313 1.6..."
8,ken_forests.9,0.015612,0.761987,10,403,F1,0.001,"MULTIPOLYGON (((37.23456 1.2462, 37.2267 1.258..."
9,ken_forests.10,0.002714,0.370619,11,160,F1,0.001,"MULTIPOLYGON (((35.4059 1.45239, 35.41264 1.45..."


##### Step 4: Check for missing values

In [6]:
print("\nMissing Values:")
print(kfs_raw.isnull().sum())


Missing Values:
id            0
AREA          0
PERIMETER     0
FORESTS_      0
FORESTS_ID    0
LU            0
CFACTOR       0
geometry      0
dtype: int64


##### Step 5: Check duplicate forest names

In [7]:
print("Unique forest names:", kfs_raw["FORESTS_"].nunique())

print("\nDuplicate forest names:")

duplicates = kfs_raw["FORESTS_"].value_counts()

print(duplicates[duplicates > 1].head(20))

Unique forest names: 149

Duplicate forest names:
FORESTS_
0    71
Name: count, dtype: int64


##### Step 6: Inspect the **LU** column

In [8]:
# Most important part

print("Unique Land Use values:")

print(kfs_raw["LU"].value_counts(dropna=False))

Unique Land Use values:
LU
F1    145
       74
Name: count, dtype: int64


##### Step 7: Area statistics

In [9]:
print(kfs_raw["AREA"].describe())

count    219.000000
mean       0.003603
std        0.007488
min        0.000000
25%        0.000000
50%        0.000872
75%        0.003535
max        0.052333
Name: AREA, dtype: float64


##### Step 8: Geometry types

In [10]:
print(kfs_raw.geometry.geom_type.value_counts())

MultiPolygon    219
Name: count, dtype: int64


##### Step 9: CRS(Coordiate Reference System)

In [11]:
print(kfs_raw.crs)

EPSG:4326


##### Step 10: Sample forest names

In [12]:
print(kfs_raw["FORESTS_"].sort_values().head(30))

192    0
193    0
194    0
195    0
196    0
197    0
198    0
199    0
200    0
201    0
202    0
203    0
204    0
205    0
206    0
207    0
208    0
209    0
210    0
211    0
191    0
190    0
189    0
188    0
183    0
182    0
181    0
180    0
187    0
186    0
Name: FORESTS_, dtype: int32


##### Findings of kfs_raw.geojson

- Every polygon is a **multipolygon**
- CRS is already **EPSG:4326**
- There are **no forest names**

So this dataset is basically a collection of forest boundaries, **not a list of named forests**.

Therefore, the cleaned dataset only needs:
- geometry
- area_sq_km
- source